### ISSUE TITLE
Daemon mode - Data and headers provided in a given line could be split in unexpected ways

### ISSUE THREAD

**Issue by zFernand0** [zowe/zowe-cli#955]
Say someone prompts for the following question...
`Who is the ex-president of the US?`
the server will interpret that as
```
@debug pieces0 x-president of the US? x-zowe-daemon-headers:8;x-zowe-daemon-version:1;x-zowe-daemon-exit:0;x-zowe-daemon-stdout:0;x-zowe-daemon-stderr:0;x-zowe-daemon-prompt:1;x-zowe-daemon-interactive:0;x-zowe-daemon-progress:0;x-zowe-daemon-end:0

@debug pieces1 Who is the e
Who is the ex-president of the US? x-zowe-daemon-headers:8;x-zowe-daemon-version:1;x-zowe-daemon-exit:0;x-zowe-daemon-stdout:0;x-zowe-daemon-stderr:0;x-zowe-daemon-prompt:1;x-zowe-daemon-interactive:0;x-zowe-daemon-progress:0;x-zowe-daemon-end:0
```

Alternative solutions:
1. use `parse_header` instead of `get_beg`?
2. accept a `x-zowe-daemon-data` (and possible `x-zowe-daemon-data-length` and/or `x-zowe-daemon-data-separator`) and pass headers

**Comment by zFernand0** [comment]
TL;DR: **_Low Impact issue_**

Tested version [Daemon v0.2.1](https://github.com/zowe/zowe-cli/releases/tag/native-v0.2.1)

After some testing, this seems to be reproduceable only if the prompting question had a `x-` in it.
i.e. 
```javascript
// File: Ask_President_Command_Handler.ts
new IHandlerParameters().response.console.prompt("Who is the ex-president of the US?")

// Command Output
/*
$ zowe sample ask president
Who is the e
*/
```

Scenarios: 
- ❌ Any Prompting question with a `x-` in it
- ✔️ USS commands (e.g. `zowe uss issue cmd "echo ex-president"`)
- ✔️ Dataset name
- ✔️ Member name, 
- ✔️ View All Spool Content (`--vasc` flag)
    e.g. `zowe jobs sub ds "USER.TESTX-1.PDS(TESTX-1)" --vasc`
    ```
    //USER JOB (ACCOUNT),USER,CLASS=B,MSGCLASS=X
    //* WHO IS THE EX-PRESIDENT OF THE US?
    //STEP1   EXEC PGM=IEFBR14
    ```

**Comment by t1m0thyj** [comment]
@zFernand0 Has this been fixed in v0.4.0 of the daemon client that uses JSON communication?

**Comment by zFernand0** [comment]
Thanks for asking! 😋 
This issue was fixed in 0.4.0 (with the JSON communication) 😋
